crates/krator/src/admission.rs

```diff
@@ -0,0 +1,115 @@
+//! Basic implementation of Kubernetes Admission API
+use crate::Operator;
+use k8s_openapi::apimachinery::pkg::apis::meta::v1::Status;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+
+/// Result of admission hook.
+#[allow(clippy::large_enum_variant)]
+pub enum AdmissionResult<T> {
+    /// Permit the request. Pass the object (with possible mutations) back.
+    /// JSON Patch of any changes will automatically be created.
+    Allow(T),
```

Comment:
Does `T` need to be `Serializable` as it is going into a JSON patch?

Reply:
Yes, but creating the actual patch only happens in the `endpoint` method so I think it makes sense to just rely on the fact that `T` comes from `Operator::Manifest`. This type isn't really intended to be used outside of the context of our API.